server/graphql/v2/mutation/OrderMutations.js

```diff
@@ -315,6 +320,42 @@ const orderMutations = {
           throw new ValidationFailed(`Only pending/expired orders can be marked as paid, this one is ${order.status}`);
         }
 
+        if (args?.details) {
+          const { totalAmount, paymentProcessorFeesAmount, platformTipAmount } = args.details;
+          const host = await toAccount.getHostCollective();
+          if (totalAmount) {
+            if (totalAmount?.currency !== order.currency) {
+              throw new ValidationFailed('totalAmount currency must match expense currency');
+            }
+            order.set('totalAmount', totalAmount.valueInCents || floatAmountToCents(totalAmount.value));
+          }
+          if (paymentProcessorFeesAmount) {
+            if (paymentProcessorFeesAmount?.currency !== host.currency) {
+              throw new ValidationFailed('paymentProcessorFeesAmount currency must match host currency');
+            }
+            if (!order.data) {
+              order.set('data', {});
+            }
+            order.set(
+              'data.paymentProcessorFeeInHostCurrency',
+              paymentProcessorFeesAmount.valueInCents || floatAmountToCents(paymentProcessorFeesAmount.value),
+            );
+          }
+          if (platformTipAmount) {
+            if (platformTipAmount?.currency !== order.currency) {
+              throw new ValidationFailed('platformTipAmount currency must match order currency');
+            }
+            if (!order.data) {
+              order.set('data', {});
+            }
+            order.set(
+              'data.platformTip',
```

Comment:
Correct me if I am wrong, but my understanding is that this should be `platformFee` not `platformTip` as that is what we normally store within the `data` object. It's kinda confusing. 🤔 

```suggestion
              'data.platformFee',
```

Reply:
There is a forward compatibility utility `getPlatformTip` that take care of both form https://github.com/opencollective/opencollective-api/blob/main/server/lib/payments.js#L787

We ultimately want to use only `platformTip` and migrate everything, but whatever that works for now.

Reply:
@SudharakaP we're moving the platform away from platform fees towards platform tips. The exception is still OSC, but in that case, I suggest we just hide the field in the frontend.

Platform-tip is working.